From 44cb70cc1c9ac94a179d6f8bd6d189dc568d741a Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Thu, 28 Mar 2024 15:10:48 +0530
Subject: [PATCH 1081/1141] arm64: dts: st: Enable Superspeed support for
 USB3DR in stm32mp257f-dk

Add the required node(s) for USB3-DWC3 Superspeed support : COMBOPHY
Add combo-phy handle in USB3DR node
Enable usb-role-switch on stm32mp257f-dk, to support dual role USB2
with power delivery on Type-C connector.
There's a dependency on the type-c connector controller, to enable
and manage the super speed switch (orientation).

Change-Id: Ic556b8effb9854b50b8793f9572f544c704c721a
Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts | 24 +++++++++++++++++++++--
 1 file changed, 22 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
index 6630e21515f0..e7d1d4e1dc45 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -184,6 +184,11 @@ &arm_wdt {
 	status = "okay";
 };
 
+&combophy {
+	st,ssc-on;
+	status = "okay";
+};
+
 &crc {
 	status = "okay";
 };
@@ -546,6 +551,16 @@ typec@35 {
 			reg = <0x35>;
 			interrupts-extended = <&intc_rpmsg 0>;
 			status = "okay";
+			connector {
+				compatible = "usb-c-connector";
+				label = "USB-C";
+
+				port {
+					typec_ep: endpoint {
+						remote-endpoint = <&dwc3_ep>;
+					};
+				};
+			};
 		};
 	};
 };
@@ -713,8 +728,13 @@ &usb3dr {
 	status = "okay";
 
 	dwc3: usb@48300000 {
-		maximum-speed = "high-speed";
-		role-switch-default-mode = "peripheral";
+		phys            = <&usb2_phy2>, <&combophy PHY_TYPE_USB3>;
+		phy-names       = "usb2-phy", "usb3-phy";
 		usb-role-switch;
+		port {
+			dwc3_ep: endpoint {
+				remote-endpoint = <&typec_ep>;
+			};
+		};
 	};
 };
-- 
2.39.2

